code/game/machinery/computer/buildandrepair.dm

```diff
@@ -138,8 +139,28 @@
 /obj/structure/frame/computer/deconstruct(disassembled = TRUE)
 	if(!(flags_1 & NODECONSTRUCT_1))
 		if(state == 4)
-			new /obj/item/shard(loc)
-			new /obj/item/shard(loc)
+			new /obj/item/shard(drop_location())
+			new /obj/item/shard(drop_location())
 		if(state >= 3)
-			new /obj/item/stack/cable_coil(loc , 5)
-	..()
\ No newline at end of file
+			new /obj/item/stack/cable_coil(drop_location(), 5)
+	..()
+
+/obj/structure/frame/computer/verb/rotate()
+	set name = "Rotate Computer Frame"
+	set category = "Object"
+	set src in oview(1)
+
+	if(usr.incapacitated())
+		return
+	
+	if(anchored)
+		to_chat(usr, "<span class='warning'>You must unwrench [src] before rotating it!</span>")
+		return
+
+	setDir(turn(dir, -90))
+
+/obj/structure/frame/computer/AltClick(mob/user)
+	..()
+	if(!in_range(src, user))

```

Comment:
this should be in rotate itself - also, this handles adjacency and ghosts, right?

Reply:
It's pretty much a copy-paste of how all other AltClick()'s work.